### PR TITLE
python311Packages.skops:  conditionally disable unit test that segfaults on Darwin

### PR DESCRIPTION
**Pull request by sarahec**
## Description of changes

`python311Pakages.skops` segfaults while testing persistance on Darwin. This patch disables that test for Darwin builds.

cc: @NixOs/darwin-maintainers

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/skops/default.nix [hunk 1]
```diff
@@ -44,13 +45,18 @@ buildPythonPackage rec {
...
+      # try to download data from Huggingface Hub:
+      "skops/hub_utils/tests"
+      "skops/card/tests"
+      # minor output formatting issue
+      "skops/card/_model_card.py"
+    ]
+    ++ lib.optional stdenvNoCC.isDarwin [
+      # Segfaults on darwin
+      "skops/io/tests/test_persist.py"
+    ];
```
Comment (prusnak) [inline pkgs/development/python-modules/skops/default.nix:59]: - use optionals
- use stdenv

```suggestion
    ++ lib.optionals stdenv.isDarwin [
      # Segfaults on darwin
      "skops/io/tests/test_persist.py"
    ];
```

File: pkgs/development/python-modules/skops/default.nix [hunk 2]
```diff
@@ -10,6 +10,7 @@
   matplotlib,
   pandas,
   scikit-learn,
+  stdenvNoCC,
```
Comment (prusnak) [inline pkgs/development/python-modules/skops/default.nix:13]: nah, we already pull in stdenv via buildPythonPackage
```suggestion
```

**Review by prusnak — CHANGES_REQUESTED** [review]

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 340596` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.skops</li>
    <li>python311Packages.skops.dist</li>
    <li>python312Packages.skops</li>
    <li>python312Packages.skops.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/skops/default.nix (hunk 2 above)
Comment (sarahec) [reply pkgs/development/python-modules/skops/default.nix]: Done. I learn something new with each PR :)

File: pkgs/development/python-modules/skops/default.nix (hunk 1 above)
Comment (sarahec) [reply pkgs/development/python-modules/skops/default.nix]: Fixed.

File: pkgs/development/python-modules/skops/default.nix (hunk 2 above)
Comment (sarahec) [reply pkgs/development/python-modules/skops/default.nix]: Got it

Comment (sarahec) [reply pkgs/development/python-modules/skops/default.nix]: Apparently it won't parse withut the explicit import. (ofBorg just failed it for that reason).

**Comment by sarahec** [comment]
@prusnak when I removed `stdenv,` from the inputs, ofBorg reported a syntax error in `    ++ lib.optionals stdenv.isDarwin [`. Perhaps `buildPythonPackage` doesn't pass it through?

**Comment by emilazy** [comment]
I think the intended meaning was “there’s no point to using `stdenvNoCC` rather than `stdenv` because the machinery this package uses already depends on the full `stdenv`”, not that you can omit the argument entirely.

**Comment by sarahec** [comment]
@emilazy so it seems. I am but an egg.

**Comment by sarahec** [comment]
Result of `nixpkgs-review pr 340596` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.skops</li>
    <li>python311Packages.skops.dist</li>
    <li>python312Packages.skops</li>
    <li>python312Packages.skops.dist</li>
  </ul>
</details>

**Review by prusnak — APPROVED** [review]

**Pull request merged**